### PR TITLE
Improving ContainerException message

### PR DESCRIPTION
**Pull request by msmakouz**
| Q             | A
| ------------- | ---
| Bugfix?       | ❌
| Breaks BC?    | ❌ 
| New feature?  | ✔️

Error message example:

```
Can't resolve `Spiral\Tests\Core\Fixtures\WithContainerInside`: undefined class or binding `invalid`.
Container trace list:
- action: 'autowire'
  alias: 'Spiral\Tests\Core\Fixtures\WithContainerInside'
  context: NULL
- call: 'Spiral\Tests\Core\Fixtures\WithContainerInside::__construct'
  arguments: [
    0: instance of Spiral\Core\Container
  ]
  - action: 'autowire'
    alias: 'invalid'
    context: NULL
```

```
Can't resolve `Spiral\Tests\Core\Fixtures\ClassWithUndefinedDependency`: undefined class or binding `invalid`.
Container trace list:
- action: 'autowire'
  alias: 'Spiral\Tests\Core\Fixtures\ClassWithUndefinedDependency'
  context: NULL
- action: 'resolve arguments'
  signature: function (Spiral\Tests\Core\Fixtures\InvalidClass $class)
  - action: 'resolve from binding'
    alias: 'Spiral\Tests\Core\Fixtures\InvalidClass'
    context: 'class'
    binding: [
      0: static function (Psr\Container\ContainerInterface $container),
      1: false
    ]
    - action: 'autowire'
      alias: 'invalid'
      context: NULL
```

### CODE REVIEW

**Comment by roxblnfk** [comment]
I think, a list will look better here

Markdown list
```
Undefined class or binding `Spiral\Http\ErrorHandler\RendererInterface`.
Container stack trace:
- Spiral\Http\Middleware\ErrorHandlerMiddleware
- Spiral\Http\Middleware\ErrorHandlerMiddleware\SuppressErrorsInterface
- Spiral\Http\Middleware\ErrorHandlerMiddleware\EnvSuppressErrors
- Spiral\Boot\Environment\DebugMode
- Spiral\Http\ErrorHandler\RendererInterface
```
Markdown numeric list
```
Undefined class or binding `Spiral\Http\ErrorHandler\RendererInterface`.
Container stack trace:
1. Spiral\Http\Middleware\ErrorHandlerMiddleware
2. Spiral\Http\Middleware\ErrorHandlerMiddleware\SuppressErrorsInterface
3. Spiral\Http\Middleware\ErrorHandlerMiddleware\EnvSuppressErrors
4. Spiral\Boot\Environment\DebugMode
5. Spiral\Http\ErrorHandler\RendererInterface
```

Error stack style
```
Undefined class or binding `Spiral\Http\ErrorHandler\RendererInterface`.
Container stack trace:
#0 Spiral\Http\Middleware\ErrorHandlerMiddleware
#1 Spiral\Http\Middleware\ErrorHandlerMiddleware\SuppressErrorsInterface
#2 Spiral\Http\Middleware\ErrorHandlerMiddleware\EnvSuppressErrors
#3 Spiral\Boot\Environment\DebugMode
#4 Spiral\Http\ErrorHandler\RendererInterface
```

File: src/Core/src/Internal/Factory.php [hunk 1]
```diff
@@ -82,6 +85,7 @@ public function make(string $alias, array $parameters = [], string $context = nu
         } finally {
             /** @psalm-var class-string $alias */
             $this->state->bindings[$alias] = $binding;
+            $this->trace = [];
```
Comment (roxblnfk) [inline src/Core/src/Internal/Factory.php:88]: What about case when the `FactoryInterface` or `ContainerInterface` are used in a dependency constructor?

File: src/Core/src/Exception/Container/ContainerException.php [hunk 2]
```diff
@@ -12,4 +12,30 @@
...
+        int $code = 0,
+        ?\Throwable $previous = null,
+        protected array &$trace = []
+    ) {
+        parent::__construct($this->addTrace($message), $code, $previous);
+
+        $trace = [];
+    }
+
+    protected function addTrace(string $message): string
```
Comment (roxblnfk) [inline src/Core/src/Exception/Container/ContainerException.php:26]:
```suggestion
    protected function generateTrace(string $message): string
```

Comment (msmakouz) [reply src/Core/src/Exception/Container/ContainerException.php]: Renamed

**Review by butschster — APPROVED** [review]

**Review by roxblnfk — APPROVED** [review]

**Pull request merged**